src/extensions/AssetMigration.sol

```diff
@@ -0,0 +1,250 @@
+    _whitelist(recipients, tokens);
+  }
+
+  /**
+   * @dev Renounces the migrator role.
+   */
+  function renounceMigrator() external onlyMigrator {
+    _setMigrator(address(0));
+  }
+
+  /**
+   * @dev Get all whitelisted addresses for the given tokens.
+   */
+  function getWhitelistedAddresses(
+    address[] calldata tokens
+  ) external view returns (address[] memory whitelisteds) {
+    AssetMigrationStorage storage $ = _getAssetMigration();
+
+    uint256 length = tokens.length;
+    whitelisteds = new address[](length);
+
+    for (uint256 i; i < length; ++i) {
+      whitelisteds[i] = $._whitelist[tokens[i]];
+    }
+  }
+
+  /**
+   * @dev Returns the pointer of the AssetMigrationStorage struct.
+   */
+  function _getAssetMigration() private pure returns (AssetMigrationStorage storage $) {
+    bytes32 loc = _$$AssetMigrationLocation();
+    require(loc != 0, ErrInvalidStorageLocation());
+
+    assembly ("memory-safe") {
+      $.slot := loc
+    }
+  }
+
+  /**
+   * @dev Returns the custom storage location of the AssetMigrationStorage struct.
+   */
+  function _$$AssetMigrationLocation() internal pure virtual returns (bytes32 storageLoc) {
+    // value is equal to keccak256(abi.encode(uint256(keccak256("ronin.bridge.AssetMigration")) - 1)) &
+    // ~bytes32(uint256(0xff))
+    storageLoc = 0x06e9d321f1aa72738d882c53ba334d30578ba3db2fbd2df66a07059b7abc9900;
+  }
+
+  /**
+   * @dev Converts the native token to its wrapped version.
+   */
+  function _wrap(
+    uint256 amount
+  ) internal returns (IERC20) {
+    IWETH w = _getWrappedNativeToken();
+    w.deposit{ value: amount }();
+
+    return IERC20(address(w));
+  }
+
+  /**
+   * @dev Sets the migrator address.
+   */
+  function _setMigrator(
+    address addr
+  ) internal {
+    _getAssetMigration()._addr = addr;
+
+    emit NewMigrator(msg.sender, addr);
+  }
+
+  /**
+   * @dev Sets the whitelist status of the recipients.
+   * This function does not revert when the recipient is already whitelisted or not.
+   * if `recipient` is zero address, it will remove the whitelist status.
+   */
+  function _whitelist(address[] calldata tokens, address[] calldata recipients) internal {
+    AssetMigrationStorage storage $ = _getAssetMigration();
+    uint256 length = recipients.length;
+
+    for (uint256 i; i < length; ++i) {
+      require(tokens[i] != _NATIVE_TOKEN_INDICATOR, ErrWhitelistWrappedTokenInstead());
+
+      $._whitelist[tokens[i]] = recipients[i];
+    }
+
+    emit WhitelistUpdated(msg.sender, tokens, recipients);
+  }
+
+  /**
+   * @dev Sets the wrapped native token.
+   */
+  function _setWrappedNativeToken(
+    address wnt
+  ) internal {
+    _getAssetMigration()._wnt = IWETH(wnt);
+
+    emit WrappedNativeTokenSet(msg.sender, wnt);
+  }
+
+  /**
+   * @dev Throws if the recipient is not whitelisted.
+   * Returns the recipient address.
+   */
+  function _requireWhitelisted(
```

Comment:
```suggestion
  function _getRecipient(
```
We usually use `_require` prefix for function that do not return any value

Reply:
renamed.